src/stages/stage3_events.c

```diff
@@ -165,30 +162,24 @@ TASK(burst_swirls, { int count; int interval; ProjPrototype *shot_proj; }) {
 // side swirls
 // typically move across a stage in a drive-by fashion
 
-TASK(side_swirl_move, { BoxedEnemy e; cmplx p0; cmplx p1; cmplx p2; } ) {
+TASK(side_swirl_move, { BoxedEnemy e; cmplx vel; cmplx accel; cmplx accel_retention; } ) {
 	Enemy *e = TASK_BIND(ARGS.e);
-	cmplx p0 = ARGS.p0;
-	cmplx p1 = ARGS.p1;
-	cmplx p2 = ARGS.p2;
-
-	for (;;) {
-		// this "move" moves in a kind of "S" shape, horizontally across the screen
-		// TODO: replicate this w/o using `for (;;)`
-		e->move = move_linear(p0 + p1 * creal(p2));
-		p2 = creal(p2) * cimag(p2) + I * cimag(p2);
-		YIELD;
-	}
+	cmplx impulse_vel = ARGS.accel * creal(ARGS.accel_retention);
+	cmplx vel = ARGS.vel;
+	real retention = cimag(ARGS.accel_retention);
+
+	e->move = move_asymptotic(impulse_vel, vel, retention);
 }
```

Comment:
This doesn't need to be a separate task anymore - just set `e->move` instead of invoking it.